Compiler panic when trying to lock `inner` of Mutex

I've encountered a panic after refactoring my program and trying to compile in an incomplete state.

This is the error I've got:
```
   Compiling playground v0.0.1 (file:///playground)
error[E0616]: field `inner` of struct `std::sync::Mutex` is private
  --> src/main.rs:10:13
   |
10 |     let _ = test.comps.inner.lock().unwrap();
   |             ^^^^^^^^^^^^^^^^

error: internal compiler error: librustc/middle/stability.rs:708: encountered unmarked API: DefId(1/0:4014 ~ std[8250]::sys_common[0]::mutex[0]::{{impl}}[1]::lock[0])
  --> src/main.rs:10:30
   |
10 |     let _ = test.comps.inner.lock().unwrap();
   |                              ^^^^

thread 'main' panicked at 'Box<Any>', librustc_errors/lib.rs:499:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0616`.

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.28.0 (9634041f0 2018-07-30) running on x86_64-unknown-linux-gnu

note: compiler flags: -C codegen-units=1 -C debuginfo=2 --crate-type bin

note: some of the compiler flags provided by cargo are hidden

error: Could not compile `playground`.

To learn more, run the command again with --verbose.
```

This is the minimal code example for reproduction:
```
use std::sync::Mutex;

struct Test {
    comps: Mutex<String>,
}

fn main() {}

fn testing(test: Test) {
    let _ = test.comps.inner.lock().unwrap();
}
```
[playground](https://play.rust-lang.org/?gist=5fc3e95c3bd42b3263895ed3efe3b978)

I've looked at a few other issues and haven't found anything similar, so hopefully this is not a duplicate.

--- a/src/test/ui/issues/issue-54062.rs
+++ b/src/test/ui/issues/issue-54062.rs
@@ -1,0 +1,13 @@
+use std::sync::Mutex;
+
+struct Test {
+    comps: Mutex<String>,
+}
+
+fn main() {}
+
+fn testing(test: Test) {
+    let _ = test.comps.inner.lock().unwrap();
+    //~^ ERROR: field `inner` of struct `std::sync::Mutex` is private
+    //~| ERROR: no method named `unwrap` found
+}

--- a/src/test/ui/issues/issue-54062.stderr
+++ b/src/test/ui/issues/issue-54062.stderr
@@ -1,0 +1,16 @@
+error[E0616]: field `inner` of struct `std::sync::Mutex` is private
+  --> $DIR/issue-54062.rs:10:13
+   |
+LL |     let _ = test.comps.inner.lock().unwrap();
+   |             ^^^^^^^^^^^^^^^^
+
+error[E0599]: no method named `unwrap` found for type `std::sys_common::mutex::MutexGuard<'_>` in the current scope
+  --> $DIR/issue-54062.rs:10:37
+   |
+LL |     let _ = test.comps.inner.lock().unwrap();
+   |                                     ^^^^^^
+
+error: aborting due to 2 previous errors
+
+Some errors have detailed explanations: E0599, E0616.
+For more information about an error, try `rustc --explain E0599`.

